src/internal/base32.ts

```diff
@@ -0,0 +1,21 @@
+export const reducedBase32 = '0123456789ABCDEFGHJKMNPQRSTVWXYZ'; // Crockford's Base32 - Excludes I, L, O, and U which may be confused with numbers
+
+/**
+ * Encodes a Date into 10 characters base32 string.
+ *
+ * @param date the Date to encode
+ */
+export const encodeDate = (date: Date): string => {
+  let now = date.getTime();
+
+  let mod;
+  let len = 10;
+  let str = '';
+  for (; len > 0; len--) {
+    mod = now % reducedBase32.length;
+    str = reducedBase32.charAt(mod) + str;
+    now = (now - mod) / reducedBase32.length;
+  }
+
+  return str;
+};
```

Comment:
I changed this to a function and inlined a few of the definitions.

```suggestion
export function encodeDate(date: Date): string {
  let value = date.valueOf();
  let result = '';
  for (let len = 10; len > 0; len--) {
    const mod = value % 32;
    result = reducedBase32.charAt(mod) + result;
    value = (value - mod) / 32;
  }

  return result;
}
```

Optional: Should we name the method `dateToBase32`?
Similar to https://github.com/faker-js/faker/pull/2936/files#diff-d1337540fbb3961f200ab34f17406efcb6c56e21f306168648f4cbd4e015cfc3R15